### PR TITLE
[clang-tidy] Improved readability-bool-conversion to be more consistent when using parentheses

### LINKED ISSUE #71852: [clang-tidy] `readability-implicit-bool-conversion` gives inconsistent suggestions on return value
The suggestions of `readability-implicit-bool-conversion` is inconsistent when parenthesis are added to the return value. 
The 2 following methods will give different suggestions even if they behave the same way.

```
int foo()
{
    return true;
}

int foo2()
{
    return (true);
}
```

Output:
```
[<source>:3:12: warning: implicit conversion bool -> 'int' [readability-implicit-bool-conversion]]
    4 |     return true;
      |            ^~~~
      |            1
[<source>:8:12: warning: implicit conversion bool -> 'int' [readability-implicit-bool-conversion]]
    9 |     return (true);
      |            ^
      |            static_cast<int>
2 warnings generated.
```

### PR DESCRIPTION
**Pull request by felix642**
Fixes #71852

### CODE REVIEW

**Comment by firewave** [comment]
Another PR for this was opened a few hours ago: #71848.

**Comment by felix642** [comment]
Hi @firewave, I think you are referencing a different issue. If I test #71852 with PR #72050 I do not get the expected behavior.

**Comment by firewave** [comment]
> Hi @firewave, I think you are referencing a different issue. If I test #71852 with PR #72050 I do not get the expected behavior.

Of course you are right. I missed there being two different issues. Sorry about the noise.

File: clang-tools-extra/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -412,6 +412,10 @@ Changes in existing checks
   do-while loops into account for the `AllowIntegerConditions` and
   `AllowPointerConditions` options.
 
+- Improved :doc:`readability-implicit-bool-conversion
```
Comment (PiotrZSL) [inline clang-tools-extra/docs/ReleaseNotes.rst:415]: merge release notes, one entry for a check, one sentence for a change

**Review by PiotrZSL — APPROVED** [review]
Description for a change (commit / PR) needed.
Except that looks fine.

**Pull request merged**